jetscape_analysis/analysis/reader/parse_ascii.py

```diff
@@ -165,7 +168,29 @@ def _parse_header_line_format_unspecified(line: str) -> HeaderInfo:
     # Compare by length first so we can short circuit immediately if it doesn't match, which should
     # save some string comparisons.
     info: Union[HeaderInfo, CrossSection]
-    if (len(values) == 19 and values[1] == "Event") or (len(values) == 17 and values[1] == "Event"):
+    if (len(values) == 17 and values[1] == "Event"):
```

Comment:
I don't think this will maintain comparability with the v2 header. The reason that the original if statement has 17 or 19 is because some value is optional (I don't recall which one). I'm also confused how you could have added values, but the expected length is now shorter?

Reply:
Actually, looking at this some more, I guess this isn't precisely compatible w/ the `_parse_header_line_format_v2` function, which isn't ideal. I know this isn't precisely your problem, but this additional makes the issue clear, so maybe you can help me figure this out? My questions:

1. Do we actually need to try to parse the unrecognized header for all cases? I guess there may be some v2 headers w/out the header version, so we need the v2 there, but that shouldn't be the case for v3. So may it doesn't need to be here at all?
2. Can we just call the decided parsers for the various headers here if we pass the if statement? I guess we run the check twice, but maybe that's a reasonable cost in the case of the version not being defined?
3. Why is the length that we pick out for the v2 parser significantly different in the two versions? Is is just a typo? Or does the v2 parser not work? (There are tests, so I guess it should work)

Reply:
Hi Raymond:
1: perhaps not - my tests indeed ran the '_parse_header_line_format_v3' part. In fact, when is the unrecognised header code ran? If it is not needed, then I could remove
2: I guess that this could be done - I could make the changes if you agree?
3. I think this is the addition of pt hat, which adds 2 characters, is this what you meant? I'll take a look at the tests when I get a chance

Reply:
You can run the existing tests with `pytest -l tests`, which **should** be enough to determine if any of the unspecified format is broken. I think the easiest path forward would be for you to undo the changes you made to the unspecified parser, we don't mess with \#2, and then we can just rely on the v3 parser for your changes